kernel/src/process/unshare.rs

```diff
@@ -0,0 +1,66 @@
+// SPDX-License-Identifier: MPL-2.0
+
+use ostd::sync::RwArc;
+
+use crate::{prelude::*, process::CloneFlags};
+
+/// Provides APIs for disassociating parts of the execution context
+pub trait ContextUnshareApi {
+    /// Unshares the file table.
+    fn unshare_files(&self);
+    /// Unshares filesystem attributes.
+    fn unshare_fs(&self);
+    /// Unshares System V semaphore.
+    fn unshare_sysvsem(&self);
+    /// Creates and enters new namespaces as specified by the `flags` argument.
+    fn unshare_namespaces(&self, flags: CloneFlags) -> Result<()>;
+}
+
+impl ContextUnshareApi for Context<'_> {
+    fn unshare_files(&self) {
+        let mut pthread_file_table = self.posix_thread.file_table().lock();
+
+        let mut thread_local_file_table_ref = self.thread_local.borrow_file_table_mut();
+        let thread_local_file_table = thread_local_file_table_ref.unwrap();
+
+        let new_file_table = RwArc::new(thread_local_file_table.read().clone());
+
+        *pthread_file_table = Some(new_file_table.clone_ro());
+        *thread_local_file_table = new_file_table;
+    }
+
+    fn unshare_fs(&self) {
+        let mut fs_ref = self.thread_local.borrow_fs_mut();
+        let new_fs = fs_ref.as_ref().clone();
+        *fs_ref = Arc::new(new_fs);
+    }
+
+    fn unshare_sysvsem(&self) {
+        // TODO: Support unsharing System V semaphore.
+        warn!("unsharing System V semaphore is not supported");
```

Comment:
This is for consistency with the `clone` implementation? Is there any reason why we should allow this (in the `clone` system call or in the `unshare` system call)?

Reply:
> This is for consistency with the clone implementation?

Yes. The unshare System V should works consistent with clone System V.

> Is there any reason why we should allow this (in the clone system call or in the unshare system call)?

This flag is widely used by clone3 in Podman and other applications, so it must not return an error. However, our current System V semaphore support is incomplete: we don't have any semaphore-related fields in process/posix-thread, so there is nothing to clone or unshare. For now we can only provide a dummy implementation.

Reply:
Thanks for the explanation!